Extensions/SkeletonObject/Gskeletonruntimeobject.js

```diff
@@ -0,0 +1,472 @@
+    if(slot){
+        slot.setZ(z);
+    }
+};
+
+gdjs.SkeletonRuntimeObject.prototype.isPointInsideSlot = function(slotPath, x, y){
+    var hitBoxes = this.getPolygons(slotPath);
+    for(var i = 0; i < this.hitBoxes.length; ++i) {
+       if ( gdjs.Polygon.isPointInside(hitBoxes[i], x, y) )
+            return true;
+    }
+
+    return false;
+};
+
+// Extension instructions
+gdjs.SkeletonRuntimeObject.prototype.raycastSlot = function(slotPath, x, y, angle, dist, closest){
+    var objW = this.getWidth();
+    var objH = this.getHeight();
+    var diffX = this.getDrawableX()+this.getCenterX() - x;
+    var diffY = this.getDrawableY()+this.getCenterY() - y;
+    var boundingRadius = Math.sqrt(objW*objW + objH*objH)/2.0;
+
+    var result = gdjs.Polygon.raycastTest._statics.result;
+    result.collision = false;
+
+    if ( Math.sqrt(diffX*diffX + diffY*diffY) > boundingRadius + dist )
+        return result;
+    
+    var endX = x + dist*Math.cos(angle*Math.PI/180.0);
+    var endY = y + dist*Math.sin(angle*Math.PI/180.0);
+    var testSqDist = closest ? dist*dist : 0;
+
+    var hitBoxes = this.getPolygons(slotPath);
+    for (var i=0; i<hitBoxes.length; i++) {
+        var res =  gdjs.Polygon.raycastTest(hitBoxes[i], x, y, endX, endY);
+        if ( res.collision ) {
+            if ( closest && (res.closeSqDist < testSqDist) ) {
+                testSqDist = res.closeSqDist;
+                result = res;
+            }
+            else if ( !closest && (res.farSqDist > testSqDist) && (res.farSqDist <= dist*dist) ) {
+                testSqDist = res.farSqDist;
+                result = res;
+            }
+        }
+    }
+    
+    return result;
+};
+
+// Warning!, assuming that gdjs.evtTools.object.twoListsTest respect object parameters order
+gdjs.SkeletonRuntimeObject.slotObjectCollisionTest = function(skl, obj, slotPath){
+    //First check if bounding circle are too far.
+    var o1w = skl.getWidth();
+    var o1h = skl.getHeight();
+    var o2w = obj.getWidth();
+    var o2h = obj.getHeight();
+
+    var x = skl.getDrawableX()+skl.getCenterX()-(obj.getDrawableX()+obj.getCenterX());
+    var y = skl.getDrawableY()+skl.getCenterY()-(obj.getDrawableY()+obj.getCenterY());
+    var obj1BoundingRadius = Math.sqrt(o1w*o1w+o1h*o1h)/2.0;
+    var obj2BoundingRadius = Math.sqrt(o2w*o2w+o2h*o2h)/2.0;
+
+    if ( Math.sqrt(x*x+y*y) > obj1BoundingRadius + obj2BoundingRadius )
+        return false;
+
+    //Do a real check if necessary.
+    var hitBoxes1 = skl.getPolygons(slotPath);
+    var hitBoxes2 = obj.getHitBoxes();
+    for(var k = 0, lenBoxes1 = hitBoxes1.length;k<lenBoxes1;++k) {
+        for(var l = 0, lenBoxes2 = hitBoxes2.length;l<lenBoxes2;++l) {
+            if ( gdjs.Polygon.collisionTest(hitBoxes1[k], hitBoxes2[l]).collision ) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+};
+
+// Warning!, assuming that gdjs.evtTools.object.twoListsTest respect object parameters order
```

Comment:
I'm not sure to understand these warnings 🤔 Could you explain a bit more?

Reply:
That's because the gdjs.sk.slotObjectCollision function makes a twoListsTest giving the skeleton objects list as first parameter and the other (common) object as second parameter, then the test is made on this function you are pointing. If the twoListsTest for any reason decides to give to this function the common object as first parameter and the skeleton second it will crash :/

But if you can ensure me that if I use twoListsTest with a list of objects A first and B second it will call the predicate with A first and B second, then I can delete this if you want :)